omod/src/main/java/org/openmrs/web/controller/user/UserFormController.java

```diff
@@ -284,6 +291,19 @@ public String handleSubmission(WebRequest request, HttpSession httpSession, Mode
 				us.changeQuestionAnswer(user, secretQuestion, secretAnswer);
 			}
 			
+			//Check if admin wants the person associated with the user to be added to the Provider Table
+			if(addToProviderTableOption != null) {
+				Set<Role> userRoles = user.getRoles();
+				for (Role role : userRoles) {
+					if (role.getRole().equals("Provider")) { //If 'Provider' role is selected, we add the User to the Provider Table
+						Provider provider = new Provider();
+						provider.setPerson(user.getPerson());
+						provider.setName(user.getName());
```

Comment:
If you have set person, you don't need to set name too, provider.getName will take care of using the name of the linked person record